### ISSUE TITLE
[UI - Dashboard/Data Explorer] Configure cell throws exception Cannot destructure property 'keys'

### ISSUE THREAD

**Issue by karel-rehor** [influxdata/influxdb#16063]
__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Generate some test data covering the past month
2. Create a Dashboard and then a cell to display the test data - save the cell so that the dashboard is shown
3. Click configure of the cell 

(For Data Explorer see further comment)
__Expected behavior:__
That the cell edit view loads without issue so that the query can be manipulated. 

__Actual behavior:__
The Flux schema navigation pane contains the message "An InfluxDB error has occured..."   I don't see any obvious error messages in the influxdb log, however console log contains. 

```
TypeError: Cannot destructure property 'keys' of 'en(...).queryBuilder.tags[t.index]' as it is undefined.
```

__Screencast__

![DashboardCellConfigurer01](https://user-images.githubusercontent.com/25881301/69806812-11f97c00-11e4-11ea-99ef-f1c9f64ae1fc.gif)

__Environment info:__

* System info: Run `uname -srm` and copy the output here
```
Linux 4.4.0-169-generic x86_64
```
* InfluxDB version: Run `influxd version` and copy the output here
This was detected in the nightly build docker image
```
ts=2019-11-28T08:48:48.770163Z lvl=info msg="Welcome to InfluxDB" log_id=0JOJ0tWW000 version=nightly commit=f27198c1a build_date=2019-11-28T07:25:07Z
```
* Other relevant environment details: Container runtime, disk info, etc

Browser: __Chrome 78__

__Config:__
Copy any non-default config values here or attach the full config as a gist or file.

Started with end to end testing enabled 
```
--e2e-testing=true
```

__Logs:__
Console Log: 
```
index.ts:470 Error
    at index.ts:116
    at :9999/async http:/localhost:9999/3.18328136f3.js:1
(anonymous) @ index.ts:470
async function (async)
(anonymous) @ index.ts:462
(anonymous) @ index.js:9
(anonymous) @ redux.js:449
Jg.handlePositionChange @ DashboardPage.tsx:254
Tg.handleLayoutChange @ Cells.tsx:132
t.onLayoutMaybeChanged @ ReactGridLayout.js:219
t.onResizeStop @ ReactGridLayout.js:324
(anonymous) @ GridItem.js:385
(anonymous) @ Resizable.js:168
ia @ react-dom.production.min.js:193
ra @ react-dom.production.min.js:193
Wa @ react-dom.production.min.js:216
Ba @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
Ps @ react-dom.production.min.js:250
Rs @ react-dom.production.min.js:249
Ls @ react-dom.production.min.js:248
Ms @ react-dom.production.min.js:245
es @ react-dom.production.min.js:243
enqueueSetState @ react-dom.production.min.js:130
_.setState @ react.production.min.js:13
(anonymous) @ Resizable.js:167
i.handleDragStop @ react-draggable.js:1010
Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:466
2react-dom.production.min.js:198 TypeError: Cannot destructure property 'keys' of 'en(...).queryBuilder.tags[t.index]' as it is undefined.
    at Function.mapToProps (TagSelector.tsx:272)
    at r (wrapMapToProps.js:41)
    at Function.r.mapToProps (wrapMapToProps.js:50)
    at r (wrapMapToProps.js:41)
    at selectorFactory.js:22
    at Object.run (connectAdvanced.js:22)
    at s.l.initSelector (connectAdvanced.js:182)
    at new Connect(dE) (connectAdvanced.js:119)
    at li (react-dom.production.min.js:132)
    at Oo (react-dom.production.min.js:167)
fa @ react-dom.production.min.js:198
Da.o.componentDidCatch.n.callback @ react-dom.production.min.js:211
ia @ react-dom.production.min.js:193
ra @ react-dom.production.min.js:193
Wa @ react-dom.production.min.js:216
Ba @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
Ps @ react-dom.production.min.js:250
Rs @ react-dom.production.min.js:249
Ls @ react-dom.production.min.js:248
As @ react-dom.production.min.js:251
Cn @ react-dom.production.min.js:85
Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:466
index.ts:734 Uncaught (in promise) TypeError: Cannot set property 'keysStatus' of undefined
    at Proxy.<anonymous> (index.ts:734)
    at e.value (immer.module.js:661)
    at Ai (index.ts:730)
    at timeMachines (index.ts:222)
    at redux.js:435
    at v (redux.js:213)
    at queryStringConfig.ts:10
    at resizeLayout.ts:7
    at middleware.js:22
    at index.js:9
(anonymous) @ index.ts:734
value @ immer.module.js:661
Ai @ index.ts:730
timeMachines @ index.ts:222
(anonymous) @ redux.js:435
v @ redux.js:213
(anonymous) @ queryStringConfig.ts:10
(anonymous) @ resizeLayout.ts:7
(anonymous) @ middleware.js:22
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:183
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:268
async function (async)
(anonymous) @ queryBuilder.ts:247
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:218
async function (async)
(anonymous) @ queryBuilder.ts:190
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:166
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:147
async function (async)
(anonymous) @ queryBuilder.ts:133
(anonymous) @ index.js:9
(anonymous) @ redux.js:449
componentDidMount @ QueryBuilder.tsx:41
Wa @ react-dom.production.min.js:216
Ba @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
Ps @ react-dom.production.min.js:250
Rs @ react-dom.production.min.js:249
Ls @ react-dom.production.min.js:248
As @ react-dom.production.min.js:251
Cn @ react-dom.production.min.js:85
Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:466
2react-dom.production.min.js:198 TypeError: Cannot destructure property 'keys' of 'en(...).queryBuilder.tags[t.index]' as it is undefined.
    at Function.mapToProps (TagSelector.tsx:272)
    at r (wrapMapToProps.js:41)
    at Function.r.mapToProps (wrapMapToProps.js:50)
    at r (wrapMapToProps.js:41)
    at selectorFactory.js:22
    at Object.run (connectAdvanced.js:22)
    at s.l.initSelector (connectAdvanced.js:182)
    at new Connect(dE) (connectAdvanced.js:119)
    at li (react-dom.production.min.js:132)
    at Oo (react-dom.production.min.js:167)
fa @ react-dom.production.min.js:198
Da.o.componentDidCatch.n.callback @ react-dom.production.min.js:211
ia @ react-dom.production.min.js:193
ra @ react-dom.production.min.js:193
Wa @ react-dom.production.min.js:216
Ba @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
Ps @ react-dom.production.min.js:250
Rs @ react-dom.production.min.js:249
Ls @ react-dom.production.min.js:248
As @ react-dom.production.min.js:251
Cn @ react-dom.production.min.js:85
Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:466
index.ts:734 Uncaught (in promise) TypeError: Cannot set property 'keysStatus' of undefined
    at Proxy.<anonymous> (index.ts:734)
    at e.value (immer.module.js:661)
    at Ai (index.ts:730)
    at timeMachines (index.ts:222)
    at redux.js:435
    at v (redux.js:213)
    at queryStringConfig.ts:10
    at resizeLayout.ts:7
    at middleware.js:22
    at index.js:9
(anonymous) @ index.ts:734
value @ immer.module.js:661
Ai @ index.ts:730
timeMachines @ index.ts:222
(anonymous) @ redux.js:435
v @ redux.js:213
(anonymous) @ queryStringConfig.ts:10
(anonymous) @ resizeLayout.ts:7
(anonymous) @ middleware.js:22
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:183
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:268
async function (async)
(anonymous) @ queryBuilder.ts:247
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:218
async function (async)
(anonymous) @ queryBuilder.ts:190
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:166
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:147
async function (async)
(anonymous) @ queryBuilder.ts:133
(anonymous) @ index.js:9
(anonymous) @ redux.js:449
componentDidMount @ QueryBuilder.tsx:41
Wa @ react-dom.production.min.js:216
Ba @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
Ps @ react-dom.production.min.js:250
Rs @ react-dom.production.min.js:249
Ls @ react-dom.production.min.js:248
As @ react-dom.production.min.js:251
Cn @ react-dom.production.min.js:85
Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:466
```

**Comment by karel-rehor** [comment]
__Also appears in Data Explorer__

1. Open Data Explorer and create an initial query to where a graph is shown 
1. Open home page
1. Open Data Explorer again 

Same issue occurs 

__Screenshot__ 

![DataExplorerCannotReload01](https://user-images.githubusercontent.com/25881301/69809170-7965fa80-11e9-11ea-8937-de4cd43e0900.png)

__Console Log__
```
TypeError: Cannot destructure property 'keys' of 'en(...).queryBuilder.tags[t.index]' as it is undefined.
    at Function.mapToProps (TagSelector.tsx:272)
    at r (wrapMapToProps.js:41)
    at Function.r.mapToProps (wrapMapToProps.js:50)
    at r (wrapMapToProps.js:41)
    at selectorFactory.js:22
    at Object.run (connectAdvanced.js:22)
    at s.l.initSelector (connectAdvanced.js:182)
    at new Connect(dE) (connectAdvanced.js:119)
    at li (react-dom.production.min.js:132)
    at Oo (react-dom.production.min.js:167)
fa @ react-dom.production.min.js:198
Da.o.componentDidCatch.n.callback @ react-dom.production.min.js:211
ia @ react-dom.production.min.js:193
ra @ react-dom.production.min.js:193
Wa @ react-dom.production.min.js:216
Ba @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
Ps @ react-dom.production.min.js:250
Rs @ react-dom.production.min.js:249
Ls @ react-dom.production.min.js:248
As @ react-dom.production.min.js:251
Cn @ react-dom.production.min.js:85
Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:466
2react-dom.production.min.js:198 TypeError: Cannot destructure property 'keys' of 'en(...).queryBuilder.tags[t.index]' as it is undefined.
    at Function.mapToProps (TagSelector.tsx:272)
    at r (wrapMapToProps.js:41)
    at Function.r.mapToProps (wrapMapToProps.js:50)
    at r (wrapMapToProps.js:41)
    at selectorFactory.js:22
    at Object.run (connectAdvanced.js:22)
    at s.l.initSelector (connectAdvanced.js:182)
    at new Connect(dE) (connectAdvanced.js:119)
    at li (react-dom.production.min.js:132)
    at Oo (react-dom.production.min.js:167)
fa @ react-dom.production.min.js:198
Da.o.componentDidCatch.n.callback @ react-dom.production.min.js:211
ia @ react-dom.production.min.js:193
ra @ react-dom.production.min.js:193
Wa @ react-dom.production.min.js:216
Ba @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
Ps @ react-dom.production.min.js:250
Rs @ react-dom.production.min.js:249
Ls @ react-dom.production.min.js:248
Ms @ react-dom.production.min.js:245
es @ react-dom.production.min.js:243
enqueueSetState @ react-dom.production.min.js:130
_.setState @ react.production.min.js:13
l.onStateChange @ connectAdvanced.js:207
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
l.onStateChange @ connectAdvanced.js:204
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
l.onStateChange @ connectAdvanced.js:204
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
l.onStateChange @ connectAdvanced.js:204
v @ redux.js:220
(anonymous) @ queryStringConfig.ts:10
(anonymous) @ resizeLayout.ts:7
(anonymous) @ middleware.js:22
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ index.ts:150
async function (async)
(anonymous) @ index.ts:148
(anonymous) @ index.js:9
(anonymous) @ redux.js:449
getResourceDetails @ GetResources.tsx:133
(anonymous) @ GetResources.tsx:105
componentDidMount @ GetResources.tsx:104
Wa @ react-dom.production.min.js:216
Ba @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
Ps @ react-dom.production.min.js:250
Rs @ react-dom.production.min.js:249
Ls @ react-dom.production.min.js:248
As @ react-dom.production.min.js:251
Cn @ react-dom.production.min.js:85
Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:466
Show 2 more frames
index.ts:734 Uncaught (in promise) TypeError: Cannot set property 'keysStatus' of undefined
    at Proxy.<anonymous> (index.ts:734)
    at e.value (immer.module.js:661)
    at Ai (index.ts:730)
    at timeMachines (index.ts:222)
    at redux.js:435
    at v (redux.js:213)
    at queryStringConfig.ts:10
    at resizeLayout.ts:7
    at middleware.js:22
    at index.js:9
(anonymous) @ index.ts:734
value @ immer.module.js:661
Ai @ index.ts:730
timeMachines @ index.ts:222
(anonymous) @ redux.js:435
v @ redux.js:213
(anonymous) @ queryStringConfig.ts:10
(anonymous) @ resizeLayout.ts:7
(anonymous) @ middleware.js:22
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:183
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:268
async function (async)
(anonymous) @ queryBuilder.ts:247
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:218
async function (async)
(anonymous) @ queryBuilder.ts:190
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:166
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ queryBuilder.ts:147
async function (async)
(anonymous) @ queryBuilder.ts:133
(anonymous) @ index.js:9
(anonymous) @ redux.js:449
componentDidMount @ QueryBuilder.tsx:41
Wa @ react-dom.production.min.js:216
Ba @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
Ps @ react-dom.production.min.js:250
Rs @ react-dom.production.min.js:249
Ls @ react-dom.production.min.js:248
Ms @ react-dom.production.min.js:245
es @ react-dom.production.min.js:243
enqueueSetState @ react-dom.production.min.js:130
_.setState @ react.production.min.js:13
l.onStateChange @ connectAdvanced.js:207
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
l.onStateChange @ connectAdvanced.js:204
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
l.onStateChange @ connectAdvanced.js:204
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
l.onStateChange @ connectAdvanced.js:204
v @ redux.js:220
(anonymous) @ queryStringConfig.ts:10
(anonymous) @ resizeLayout.ts:7
(anonymous) @ middleware.js:22
(anonymous) @ index.js:9
dispatch @ redux.js:563
(anonymous) @ index.ts:150
async function (async)
(anonymous) @ index.ts:148
(anonymous) @ index.js:9
(anonymous) @ redux.js:449
getResourceDetails @ GetResources.tsx:133
(anonymous) @ GetResources.tsx:105
componentDidMount @ GetResources.tsx:104
Wa @ react-dom.production.min.js:216
Ba @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
Ps @ react-dom.production.min.js:250
Rs @ react-dom.production.min.js:249
Ls @ react-dom.production.min.js:248
As @ react-dom.production.min.js:251
Cn @ react-dom.production.min.js:85
Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:466
Show 28 more frames
```

**Comment by karel-rehor** [comment]
Note this is not happening with cells in dashboards generated from templates.

**Comment by desa** [comment]
@karel-rehor can you check to see if this is still happening? I believe it should have been resolved by e52705d2b1f3f2949d46ad8aad48c159ef54ff32

**Comment by karel-rehor** [comment]
@desa in the latest nightly build both the Dashboard and the Data Explorer use cases seem to be working normally again. 
```
ts=2019-12-05T09:12:46.226260Z lvl=info msg="Welcome to InfluxDB" log_id=0JXLAg_W000 version=nightly commit=5f19c6cac build_date=2019-12-05T07:25:48Z
```
closing
